### PR TITLE
Internalize DuplexPipe

### PR DESCRIPTION
**Pull request by davidfowl**
React to https://github.com/aspnet/KestrelHttpServer/pull/2428

### CODE REVIEW

File: build/dependencies.props [hunk 1]
```diff
@@ -12,7 +12,7 @@
     <MicrosoftAspNetCoreAuthorizationPackageVersion>2.1.0-preview2-30451</MicrosoftAspNetCoreAuthorizationPackageVersion>
     <MicrosoftAspNetCoreAuthorizationPolicyPackageVersion>2.1.0-preview2-30451</MicrosoftAspNetCoreAuthorizationPolicyPackageVersion>
     <MicrosoftAspNetCoreBenchmarkRunnerSourcesPackageVersion>2.1.0-preview2-30451</MicrosoftAspNetCoreBenchmarkRunnerSourcesPackageVersion>
-    <MicrosoftAspNetCoreConnectionsAbstractionsPackageVersion>2.1.0-preview2-30451</MicrosoftAspNetCoreConnectionsAbstractionsPackageVersion>
+    <MicrosoftAspNetCoreConnectionsAbstractionsPackageVersion>2.1.0-a-preview2-remove-duplex-pipe-17616</MicrosoftAspNetCoreConnectionsAbstractionsPackageVersion>
```
Comment (JamesNK) [inline build/dependencies.props]: Test this doesn't break VS build.

Comment (davidfowl) [reply build/dependencies.props]: LOL 😄

Comment (davidfowl) [reply build/dependencies.props]: image

Comment (anurse) [reply build/dependencies.props]: Are we going to have to go back to the days of yore, where we had to have our code checkouts in `C:\dd` in order to keep paths short? 😛

File: src/Common/DuplexPipe.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            var input = new Pipe(inputOptions);
+            var output = new Pipe(outputOptions);
+
+            var transportToApplication = new DuplexPipe(output.Reader, input.Writer);
+            var applicationToTransport = new DuplexPipe(input.Reader, output.Writer);
+
+            return new DuplexPipePair(applicationToTransport, transportToApplication);
+        }
+
+        // This class exists to work around issues with value tuple on .NET Framework
```
Comment (anurse) [inline src/Common/DuplexPipe.cs]: μ-nit: It's not a class though 🙃

**Pull request merged**